bundles/org.eclipse.jface.text/src/org/eclipse/jface/text/codemining/LineHeaderCodeMining.java

```diff
@@ -73,4 +77,27 @@ public LineHeaderCodeMining(Position position, ICodeMiningProvider provider, Con
 		super(position, provider, action);
 	}
 
+	@Override
+	public Point draw(GC gc, StyledText textWidget, Color color, int x, int y) {
+		String title= getLabel() != null ? getLabel() : "no command"; //$NON-NLS-1$
+		String[] lines= title.split("\n"); //$NON-NLS-1$
+		if (lines.length > 1) {
+			Point result= null;
```

Comment:
This is a bit more roundabout than it needs to be (took me 2 minutes to understand vs 10 seconds) - how about directly initializing result with 					
`result= gc.stringExtent(lines[0]);` and then iterating over the rest of the lines (which in most annotations don't exist) with ``for (String line: lines.subList(1, lines.size()))`

Reply:
Could you please review the changed code in [fa63717]? I am not quite happy with the changes, I hear the clean code guys screaming "do not repeat yourself". Maybe I misunderstood your request?

Reply:
I was hoping we could remove some extra logic inside of the loop to make it 1) shorter 2) less cyclomatically complex

```java
		Point result= gc.stringExtent(lines[0]); // we know there is at least one line!)
			for (String line: lines.subList(1, lines.size())) {
				line= line.trim();
				gc.drawString(line, x, y, true);
				Point ext= gc.stringExtent(line);
				result.x= Math.max(result.x, ext.x);
				result.y+= ext.y;
				y+= result.y + textWidget.getLineSpacing();
			}
```

I make the assumption that gc.stringExtent never returns null. If it does in some edge-cases, my proposal is not equivalent to yours anymore

Reply:
Following parts are missing in your proposal:
. gc.drawString is not called for the first line
- y is not incremented with the height of the first line
- variable lines is an old fashioned array and not a list to call subList

my very latest proposal in 8884099 removes the if and does not duplicate the calls gc.drawString. 
```
Point result= new Point(0, 0);
for (String line : lines) {
	gc.drawString(line, x, y, true);
	Point ext= gc.stringExtent(line);
	result.x= Math.max(result.x, ext.x);
	result.y+= ext.y;
	y+= result.y + textWidget.getLineSpacing();
}
```

Reply:
Looks good! Sorry for the incomplete proposal